evm/evm-processor/src/ds-rpc/filter.ts

```diff
@@ -140,6 +145,16 @@ export function filterBlock(block: Block, dataRequest: DataRequest): void {
             if (rel.transaction) {
                 include.addTransaction(log.transaction)
             }
+            if (rel.transactionLogs && log.transaction) {
```

Comment:
archive also doesn't include transaction if it is not explicitly requested with `log.transaction`

Reply:
`.transactionLogs` and `.transactionTraces` don't imply transaction presence. Hence, log inclusion should not rely on `.transaction` property. 

`groupBy(logs, log => log.transactionIndex)` should be used instead.